java/flight/src/main/java/org/apache/arrow/flight/NoOpFlightProducer.java

```diff
@@ -45,8 +41,8 @@ public FlightInfo getFlightInfo(CallContext context,
   }
 
   @Override
-  public Callable<PutResult> acceptPut(CallContext context,
-      FlightStream flightStream) {
+  public Runnable acceptPut(CallContext context,
```

Comment:
does it make sense to do some kind of Cancellable type thing here?

Reply:
Do you mean being able to cancel an ongoing put from the server-side? Yes, we should expose that as well.